genutil/common_test.go

```diff
@@ -651,3 +652,254 @@ func TestFindChildren(t *testing.T) {
+				Type:   &yang.YangType{},
+				Config: yang.TSFalse,
+				Dir: map[string]*yang.Entry{
+					"type": {
+						Name:   "type",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type: &yang.YangType{
+							Kind: yang.Ystring,
+						},
+					},
+					"name": {
+						Name:   "name",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type:   &yang.YangType{Kind: yang.Ystring},
+					},
+					"admin-status": {
+						Name:   "admin-status",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type:   &yang.YangType{Kind: yang.Ystring},
+					},
+				},
+			},
+			"name": {
+				Name:   "name",
+				Kind:   yang.LeafEntry,
+				Config: yang.TSTrue,
+				Type: &yang.YangType{
+					Kind: yang.Yleafref,
+					Path: "../config/name", // To be modified within test.
+				},
+			},
+		},
+	}
+	inEntry = &yang.Entry{
+		Name: "container",
+		Kind: yang.DirectoryEntry,
+		Type: &yang.YangType{},
+		Dir: map[string]*yang.Entry{
+			"choice": {
+				Name: "choice",
+				Kind: yang.ChoiceEntry,
+				Type: &yang.YangType{},
+				Dir: map[string]*yang.Entry{
+					"case": {
+						Name: "case",
+						Kind: yang.CaseEntry,
+						Type: &yang.YangType{},
+						Dir: map[string]*yang.Entry{
+							"interface": inEntry,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	wantEntry := &yang.Entry{
+		Name:     "interface",
+		Kind:     yang.DirectoryEntry,
+		Type:     &yang.YangType{},
+		ListAttr: &yang.ListAttr{},
+		Dir: map[string]*yang.Entry{
+			"config": {
+				Name:   "config",
+				Kind:   yang.DirectoryEntry,
+				Type:   &yang.YangType{},
+				Config: yang.TSTrue,
+				Dir: map[string]*yang.Entry{
+					"type": {
+						Name:   "type",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSTrue,
+						Type: &yang.YangType{
+							Kind: yang.Ystring,
+						},
+					},
+					"name": {
+						Name:   "name",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSTrue,
+						Type: &yang.YangType{
+							Kind: yang.Ystring,
+						},
+					},
+				},
+			},
+			"state": {
+				Name:   "state",
+				Kind:   yang.DirectoryEntry,
+				Type:   &yang.YangType{},
+				Config: yang.TSFalse,
+				Dir: map[string]*yang.Entry{
+					"type": {
+						Name:   "type",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type: &yang.YangType{
+							Kind: yang.Ystring,
+						},
+					},
+					"name": {
+						Name:   "name",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type:   &yang.YangType{Kind: yang.Ystring},
+					},
+					"admin-status": {
+						Name:   "admin-status",
+						Kind:   yang.LeafEntry,
+						Config: yang.TSFalse,
+						Type:   &yang.YangType{Kind: yang.Ystring},
+					},
+				},
+			},
+			"name": {
+				Name:   "name",
+				Kind:   yang.LeafEntry,
+				Config: yang.TSTrue,
+				Type: &yang.YangType{
+					Kind: yang.Yleafref,
+					Path: "../config/name", // To be modified within test.
+				},
+			},
+		},
+	}
+	wantEntry = &yang.Entry{
```

Comment:
It might be best to call this something different than `wantEntry` -- it wasn't immediately clear to me that we were specifying `wantEntry` within this new `wantEntry` maybe the one above can just be called `childEntry`?

Reply:
I'd maybe even just call this `schemaTemplate` - just because we mutate it in the test code.